src/reflection.js

```diff
@@ -1,7 +1,18 @@
 import { filter } from 'funcadelic';
 
+const { getPrototypeOf, getOwnPropertyDescriptors, assign } = Object;
+
 export function methodsOf(Type) {
   return filter(({ key: name, value: desc }) => {
     return name !== 'constructor' && typeof name === 'string' && typeof desc.value === 'function';
-  }, Object.getOwnPropertyDescriptors(Type.prototype));
+  }, getPrototypeDescriptors(Type));
 }
+
+export function getPrototypeDescriptors(constructor) {
+  let prototype = getPrototypeOf(constructor);
```

Comment:
I'm actually confused how this works since `Type` is the constructor itself.

```js
class Foo {}
Object.getPrototypeOf(Foo) === Foo.prototype //=> false
Object.getPrototypeOf(Foo) === Function.prototype //=> true
```

Reply:
@cowboyd I pushed comment to the code to show you how it works.